fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt

```diff
@@ -64,18 +69,21 @@ class FulladlePlugin : Plugin<Project> {
             }
           }
           pluginManager.withPlugin("com.android.library") {
-            val library = extensions.getByType<LibraryExtension>()
-            library.testVariants.configureEach {
-              if (file("$projectDir/src/androidTest").exists()) {
-                outputs.configureEach {
-                  flankGradleExtension.additionalTestApks.add(
-                    root.provider {
-                      "- test: $outputFile"
-                    }
-                  )
+            val fulladleModuleExtension = extensions.getByType(FulladleModuleExtension::class.java)
```

Comment:
is `extension. getByType()` a good idea? `getByType` will throw exception if a module doesn't have a `FulladleModuleExtension`. I don't know if we should force modules to override values.

Reply:
It'll be added to all subprojects on line 21. Is there a better way to do this?

Reply:
How about [`findByType()`](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/ExtensionContainer.html#findByType-java.lang.Class-)? It won't throw an exception if a module doesn't provide the extension, and you won't have to add it to every module.

Reply:
It's a good point, but I would not expect the consumer of the plugin to create the extension when they need it. In the same way that I would not expect the consumer of the Android Gradle Plugin to create the Android extension when they need it. It should be created in each module so that it can be easily configured.
So instead of devs having to write:
`extensions.create("fulladleModuleConfig", FulladleModuleExtension::class.java) {`
thet can instead write:
`fulladleModuleConfig {`

Is that what you mean? or are we discussing different things?

Reply:
Makes sense, better to add it to every module and let consumers use `fulladleModuleConfig {}`.